app/src/main/java/com/palone/planahead/services/alarms/AlarmBroadcastReceiver.kt

```diff
@@ -44,22 +51,43 @@ class AlarmBroadcastReceiver : BroadcastReceiver() {
             context?.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
         notificationManager.createNotificationChannel(channel)
 
-        val singleNotification = NotificationCompat.Builder(context, channelId)
-            .setContentTitle(task?.description)
-            .setSmallIcon(R.drawable.ic_launcher_foreground)
-            .setContentText("Nie zrobiłeś jeszcze tej rzeczy!").build()
-        val persistentNotification = NotificationCompat.Builder(context, channelId)
+        val markAsDoneIntent = Intent(context, MarkAsDoneService::class.java)
+        markAsDoneIntent.putExtra("task", task)
+        markAsDoneIntent.putExtra("alert", alert)
+        markAsDoneIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        markAsDoneIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
+
+
+        val markAsDonePendingIntent = PendingIntent.getService(
+            context,
+            alert?.alertId ?: 0,
+            markAsDoneIntent,
+            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+        )
+
+        val notification = NotificationCompat.Builder(context, channelId)
             .setContentTitle(task?.description)
             .setSmallIcon(R.drawable.ic_launcher_foreground)
-            .setOngoing(true)
-            .setContentText("Nie zrobiłeś jeszcze tej rzeczy!").build()
+            .setContentText("You haven't done that yet!")
+            .setCategory(NotificationCompat.CATEGORY_ALARM)
+            .addAction(R.drawable.ic_launcher_foreground, "Done", markAsDonePendingIntent).apply {
+                if (alert?.alertTypeName == AlertType.ALARM) {
+                    setFullScreenIntent(alarmScreenPendingIntent(context, alert, task), true)
+                }
+            }
+
+
```

Comment:
Revert logic to remove nesting:

```
if (alert == null) {
    return;
}
```

Or even better - it seems that if `alert == null`, this method will do nothing. In that case, just change `alert` param to by not nullable and check whether it's null in the calling method instead.